### ISSUE TITLE
[Feature Request] Supporting JPMS

### ISSUE THREAD

**Issue by arjenzhou** [ClickHouse/clickhouse-jdbc#713]
Since Java 9 introduces JPMS, libs without `module-info` or Automatic module declaration cannot be used after Java 9, is there any plan to support it?

**Comment by arjenzhou** [comment]
related discussions: 
https://github.com/brettwooldridge/HikariCP/issues/1045
https://github.com/oshi/oshi/pull/1247

**Comment by zhicwu** [comment]
Thank you @arjenzhou. Yes, we definitely need this.

**Comment by zhicwu** [comment]
Looks like JPMS is still not widely adopted. `Automatic-Module-Name` is merely a workaround to make things work the old way. I tried multiple profiles(jre8 & jre11) in Maven, but it's difficult to release multiple jars in one go, not to mention we had to exclude different files in different profiles. As a result, I switched to multi-release jar ([JEP#238](https://openjdk.java.net/jeps/238), which is less flexible but a better way to manage source code and deployment. However, we still have to repackage some third party libraries like RoaringBitmap, gRPC and maybe more :<
